File: AiStudio4/AiStudio4.Web/src/types/theme.ts
// AiStudio4.Web/src/types/theme.ts

export interface Theme {
  guid: string;
  name: string;
  description: string;
  author?: string;
  previewColors: string[];
  themeJson: Record<string, Record<string, string>>;
  created: string;
  lastModified: string;
}

export interface ThemeCategory {
  name: string;
  colors: Record<string, string>;
}

export interface ThemeColors {
  [category: string]: Record<string, string>;
}

File: AiStudio4/Core/Models/Theme.cs

using System.Collections.Generic;

namespace AiStudio4.Core.Models
{
    
    
    
    public class Theme
    {
        
        
        
        public string Guid { get; set; }

        
        
        
        public string Name { get; set; }

        
        
        
        public string Description { get; set; }

        
        
        
        public string Author { get; set; }

        
        
        
        public List<string> PreviewColors { get; set; }

        
        
        
        public object ThemeJson { get; set; }

        
        
        
        public string FontCdnUrl { get; set; }

        
        
        
        public string Created { get; set; }

        
        
        
        public string LastModified { get; set; }
    }
}
